chord_drs/routes.py

```diff
@@ -121,6 +122,26 @@ def object_info(object_id):
     return jsonify(response)
 
 
+@drs_service.route('/search', methods=['GET'])
+def object_search():
+    response = []
+    name = request.args.get('name', None)
+
+    if name:
+        # if it includes a dot, we do a strict search.
```

Comment:
does it make more sense to split this endpoint into a fuzzy search vs a precise name search?
also, we'll have to deal with duplicate names on ingest at some point (right now it'll cause an error with the File backend) - are users going to be querying a deduplicated real-filesystem name or the original name of what they uploaded? (probably the latter makes more sense)

Reply:
Ok so for the first part, it's true the behavior of the endpoint is opaque as it is, so I've made it so I read from two different GET param, either name or fuzzy_name. It's unclear how this search feature will evolve in the future so I wanted to keep it smallish, and now the behavior should be clearer.

And I've implemented some logic for duplicate names - the location field should be now unique since it's appended with part of the ID. The name field in itself stays the same. I still expect that when a service feeds a file into DRS, it should keep track for the ID on its end. Since there is a possibility that when you search, there could be multiple objects with the same name (which is the same for a normal filesystem, we have file with the same name all over the place, in different locations).